From: Philip Withnall <withnall@endlessm.com>
Date: Mon, 27 Nov 2017 14:14:07 +0000
Subject: externalwindow: Add a missing break in a switch statement

Spotted by Coverity, CID 207681.

Signed-off-by: Philip Withnall <withnall@endlessm.com>
Origin: upstream, 0.10, commit:7cfbc530da71caa6db86eebb008fce1640f92fb0
---
 src/externalwindow.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/externalwindow.c b/src/externalwindow.c
index 5858e00..9b81670 100644
--- a/src/externalwindow.c
+++ b/src/externalwindow.c
@@ -131,6 +131,7 @@ external_window_get_property (GObject    *object,
     {
     case PROP_DISPLAY:
       g_value_set_object (value, priv->display);
+      break;
 
     default:
       G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
